### PR TITLE
[NumPy] Aims and objectives

### PR DESCRIPTION
**Pull request by dkillick**
Adds learning outcomes (i.e. objectives) to each top-level section within the NumPy course. Also makes sure the section on aims at the start of the course lays out the course aims.

### CODE REVIEW

**Comment by dkillick** [comment]
@pp-mo we've discussed this idea offline already – might you be able to take a look please?

**Comment by pelson** [comment]
I like the idea. I do worry that it is distracting to the learner though. Would you consider moving these to the end of each section? For example:

```
**learning outcome**: By the end of this section you will be able to ...

<section on ...>
```

Becomes:

```
<section on ...>

**learning outcome**: You should now be able to ...
```

File: course_content/notebooks/numpy_intro.ipynb [hunk 1]
```diff
@@ -69,6 +69,9 @@
    "metadata": {},
    "source": [
     "## Getting Started with NumPy <a class=\"anchor\" id=\"getting-started\"></a>\n",
+    "\n",
+    "**Learning outcome:** by the end of this section, you will be able to create NumPy arrays and apply the contents of the NumPy documentation to arrays you have created.\n",
+    "\n",
```
[the remark this answers is not in the document]
Comment (pp-mo) [reply course_content/notebooks/numpy_intro.ipynb]: :+1: much nicer !  (I think original sounds odd)

File: course_content/notebooks/numpy_intro.ipynb [hunk 2]
```diff
@@ -1161,7 +1225,9 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Efficiency <a class=\"anchor\" id=\"efficiency\"></a>"
+    "## Efficiency <a class=\"anchor\" id=\"efficiency\"></a>\n",
+    "\n",
+    "**Learning outcome:** by the end of this section, you will be able to explain the benefits and risks of the concept of views on NumPy arrays and use vectorisation to perform fast NumPy array calculations."
```
Comment (pp-mo) [inline course_content/notebooks/numpy_intro.ipynb]: This one is a less obvious pairing of goals ?

**Comment by pp-mo** [comment]
:+1: recapping a teaching section's learning outcomes at the end of a section as well.

I like the idea of a *single* goal statement at start of each section, and a list of *several* learning points at the end.
Where the initial statements seem  "a bit compound", that could indicate overloading of the sectional approach.  For example : https://github.com/SciTools/courses/pull/92/files#r183380566

File: course_content/notebooks/numpy_intro.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (pp-mo) [reply course_content/notebooks/numpy_intro.ipynb]: This change is still desired, not yet done ...

**Comment by dkillick** [comment]
> still missing a learning outcome for the 'statistics' section

Statistics is a subsection of the Arithmetic and Broadcasting section, so doesn't need a learning outcome. Or, looking at it another way, if Statistics gained a learning outcome, then the subsections on Broadcasting, Elementwise Arithmetic, Indexing, and many others would also need learning outcomes.

> This change is still desired

Whoops! That one slipped through. I'll fix that now.

**Comment by pp-mo** [comment]
> Statistics is a subsection of the Arithmetic and Broadcasting section,

I don't think so ?

And the section is headed with '##' like all the others in the above list
And the TOC at the top has :
```
"source": [
    "## Table of Contents\n",
    "\n",
    "* [Getting Started with NumPy](#getting-started)\n",
    "* [Motivation: what are arrays good for?](#motivation)\n",
    "* [The Array Object](#array-object)\n",
    "* [Application: Arithmetic and Broadcasting](#arithmetic_and_broadcasting)\n",
    "* [Application: Statistics](#statistics)\n",
    "* [Application: Efficiency](#efficiency)\n",
    "* [Further Reading](#further_reading)"
   ]
```

Which I think that is correct + as intended.

**Comment by dkillick** [comment]
>> Statistics is a subsection of the Arithmetic and Broadcasting section,
>
> I don't think so ?

"Statistics" is a subsection... but not of the Arithmetic and Broadcasting section! I've no idea what happened to the Statistics and Masked Arrays section in the rendered preview I was working from, but I have now added a learning outcome for the (missing!) section.

> That one slipped through

Not this time! @pp-mo what do you think now?

**Comment by pp-mo** [comment]
I'm liking this structure now + I think it generally reads clearly.

**Pull request merged**